pkcs11/signer.go

```diff
@@ -38,13 +36,34 @@ import (
 	"github.com/theparanoids/crypki/x509cert"
 )
 
-/* Request holds information needed by the collector to fetch the request & process it.
-   It also has a channel on which it waits for the response.
-*/
+// Request holds information needed by the collector to fetch the request & process it.
+// It has multiple channels, one for response & other to notify the worker if the client request times out to stop
+// processing any request from the client.
 type Request struct {
-	pool       sPool                        // pool is a signer pool per identifier from which to fetch the signer
-	identifier string                       // identifier indicates the endpoint for which we are fetching the signer in order to sign it
-	respChan   chan signerWithSignAlgorithm // respChan is the channel where the worker sends the signer once it gets it from the pool
+	pool       sPool           // pool is a signer pool per identifier from which to fetch the signer
+	method     string          // method is type of method the worker needs to call based on the request
+	stop       chan bool       // stop is the channel which is closed when the consumer/client times out or cancels request
+	signerData signerMetadata  // signerMetadata is an interface which each request can override & use to sign data
+	respChan   chan []byte     // respChan is the channel where the worker sends the signed data
+	errChan    chan error      // errChan is channel where the worker sends an error in case it is not able to sign the request
+    ctx        context.Context // ctx is the request context declared at caller side.
```

Comment:
acc. to golang standards we cannot pass ctx in struct as it could have side effects which we cannot handle. ref: https://go.dev/blog/context-and-structs